### PR TITLE
Implement add-user btn

### PR DESCRIPTION
**Pull request by mgandore**

### CODE REVIEW

File: component-interaction/src/app/user-list/user-list.component.html [hunk 1]
```diff
@@ -1,4 +1,6 @@
 <h2>Users and their details:</h2>
 
 <user-details *ngFor="let user of allUsers" [user]="user" (onDeleteUser)="handleDeleteUser($event)">
-</user-details>
\ No newline at end of file
+</user-details>
+<br><br>
+<button class="success" (click)="handleAddClick()">Add User</button>
```
Comment (danielursu) [inline component-interaction/src/app/user-list/user-list.component.html]: move the button to the start of the page

File: component-interaction/src/app/user-list/user-list.component.ts [hunk 2]
```diff
@@ -47,4 +47,23 @@ export class UserListComponent implements OnInit {
 		alert(`User with Id: ${event} will be deleted`)
 		this.allUsers = this.allUsers.filter((user) => user.id !== event);
 	}
+
+	public getRandomStr(): string {
+		let randomStr = "";
+		const characters = "abcdefghijklmnopqrstuvxyz";
```
Comment (danielursu) [inline component-interaction/src/app/user-list/user-list.component.ts]:
```suggestion
	public getRandomText(): string {
		let randomText: string = "";
		const alphabetLetters: string = "abcdefghijklmnopqrstuvxyz";
		
```

File: component-interaction/src/app/user-list/user-list.component.ts [hunk 3]
```diff
@@ -47,4 +47,23 @@ export class UserListComponent implements OnInit {
...
 	}
+
+	public getRandomStr(): string {
+		let randomStr = "";
+		const characters = "abcdefghijklmnopqrstuvxyz";
+		randomStr += characters.charAt(Math.floor(Math.random() * characters.length)).toUpperCase();
+		for (let i = 0; i < 5; i++) {
+			randomStr += characters.charAt(Math.floor(Math.random() * characters.length));
+		}
+		return randomStr
```
Comment (danielursu) [inline component-interaction/src/app/user-list/user-list.component.ts]:
```suggestion
		return randomText;

```

File: component-interaction/src/app/user-list/user-list.component.ts [hunk 4]
```diff
@@ -47,4 +47,23 @@ export class UserListComponent implements OnInit {
...
+	}
+
+	public handleAddClick(): void {
+		this.allUsers.push(
+			{
+				id: Math.floor(Math.random() * 100),
+				name: this.getRandomStr(),
+				contact: this.getRandomStr()
+			})
```
Comment (danielursu) [inline component-interaction/src/app/user-list/user-list.component.ts]:
```suggestion
			});

```

**Review by danielursu — CHANGES_REQUESTED** [review]

**Review by danielursuu — APPROVED** [review]

**Pull request merged**